application/classes/Ushahidi/Console/PostExporter.php

```diff
@@ -164,25 +148,18 @@ protected function executeExport(InputInterface $input, OutputInterface $output)
 		foreach ($posts as $idx => $post) {
 			// Retrieved Attribute Labels for Entity's values
 			$post = $this->postExportRepository->retrieveMetaData($post->asArray(), $keyAttributes);
- 			$posts[$idx] = $post;
- 		}
-		/**FIXME: how to make sure header_row is null/empty instead off an array with an empty item in it? */
-		if (empty($job->header_row) || $job->header_row[0] == '') {
-
-			$header_row = service("formatter.entity.post.$format")->createHeading($attributes, $posts);
-			$job->setState(['header_row' => $header_row]);
-			//FIXME update does not exist $job->update();
-		} else {
-			service("formatter.entity.post.$format")->setHeading($job->header_row);
+			$posts[$idx] = $post;
 		}
-		$_attributes = [];
-		foreach($attributes as $key => $item)
-		{
-			$_attributes[$item['key']] = $item;
+
+		if (empty($job->header_row)) {
+			$job->setState(['header_row' => $attributes]);
+            $this->exportJobRepository->update($job);
 		}
+		$header_row = $this->formatter->createHeading($job->header_row, $posts);
+		$this->formatter->setHeading($header_row);
+
+		$file = service("formatter.entity.post.$format")->__invoke($posts, $keyAttributes);
```

Comment:
This can be `this->formatter` now right?

Reply:
although I don't see the actual injection set up yet?

Reply:
solved